plugins/parsers/csv/parser.go

```diff
@@ -96,22 +96,38 @@ func (p *Parser) compile(r io.Reader) *csv.Reader {
 
 func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	r := bytes.NewReader(buf)
+	return parseCSV(p, r)
+}
+
+// ParseLine does not use any information in header and assumes DataColumns is set
+// it will also not skip any rows
+func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
+	r := bytes.NewReader([]byte(line))
+	metrics, err := parseCSV(p, r)
+	if len(metrics) > 0 {
+		// only return the first metric as there should be only one
+		return metrics[0], err
+	}
+	return nil, err
```

Comment:
Can you please change that to check there is only _exactly_ one metric?
```suggestion
	if err != nil {
		return nil, err
	}
	if len(metrics) != 1 {
		return nil, fmt.Errorf("expected 1 metric found %d", len(metrics))
	}
	return metrics[0], nil
```